fix(linter): get cli args on JS side, to avoid runtime inconsistencies

camc314 opened the pull request:
fixes #14150

related: #14112, #14071

This seems to be the only reliable way to get arguments across different runtimes.

when running with deno, the args from `std::env::args_os()` look like:
```
[
    "/Users/cameron/.deno/bin/deno",
    "run",
    "--ext=js",
    "-A",
    "/Users/cameron/github/camc314/oxlint-repros/issue-14150/node_modules/.deno/oxlint@1.18.0/node_modules/oxlint/bin/oxlint",
    ".",
]
```

however `process.argv` looks like:
```
[
  "oxlint",
  "/Users/cameron/github/camc314/oxlint-repros/issue-14150/node_modules/.deno/oxlint@1.18.0/node_modules/oxlint/bin/oxlint",
  "."
]
```

so skipping the first two means that we end up with incorrect args that we're trying to parts.

There are (potentially) other solutions e.g. search for one of the args ending in`oxlint`, or other methods of trying to guess.

However, this feels like the most reliable solution.

camc314 commented:
* **#14223** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/14223?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈 <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/14223?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `main`
<hr />

## How to use the Graphite Merge Queue

Add either label to this PR to merge it via the merge queue:
- <kbd>0-merge</kbd> - adds this PR to the back of the merge queue
- <kbd>hotfix</kbd> - for urgent hot fixes, skip the queue and merge this PR next

You must have a Graphite account in order to use the merge queue. Sign up using [this link](https://app.graphite.dev/invite/github/oxc-project?ref=merge-queue-instructions-in-stack-comment&prId=14223).

<h2></h2>
<sub>

An organization admin has enabled the [Graphite Merge Queue](https://graphite.dev/docs/graphite-merge-queue) in this repository.

</sub>

<sub>

Please do not merge from GitHub as this will restart CI on PRs being processed by the merge queue.

</sub>

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.
<!-- Current dependencies on/for this PR: -->

overlookmotel reviewed (approved):
> however `process.argv` looks like:
> 
> ```
> [
>   "oxlint",
>   "/Users/cameron/github/camc314/oxlint-repros/issue14150/node_modules/.deno/oxlint@1.18.0/node_modules/oxlint/bin/oxlint",
>   "."
> ]
> ```

That looks really weird. Isn't that what you'd expect to get if the command being run was `oxlint oxlint .`?

But sure, if it works on all platforms...

camc314 commented:
### Merge activity

* **Sep 29, 8:32 AM UTC**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.

camc314 committed:
fix(linter): get cli args on JS side, to avoid runtime inconsistencies (#14223)

fixes #14150

related: #14112, #14071

This seems to be the only reliable way to get arguments across different runtimes.

when running with deno, the args from `std::env::args_os()` look like:
```
[
    "/Users/cameron/.deno/bin/deno",
    "run",
    "--ext=js",
    "-A",
    "/Users/cameron/github/camc314/oxlint-repros/issue-14150/node_modules/.deno/oxlint@1.18.0/node_modules/oxlint/bin/oxlint",
    ".",
]
```

however `process.argv` looks like:
```
[
  "oxlint",
  "/Users/cameron/github/camc314/oxlint-repros/issue-14150/node_modules/.deno/oxlint@1.18.0/node_modules/oxlint/bin/oxlint",
  "."
]
```

so skipping the first two means that we end up with incorrect args that we're trying to parts.

There are (potentially) other solutions e.g. search for one of the args ending in`oxlint`, or other methods of trying to guess.

However, this feels like the most reliable solution.

camc314 commented:
lets release with this change, and get some feedback from people using this in the real world
